src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const obj = { ...state };
+  const arr = [];
+
+  for (const char of actions) {
+    switch (char.type) {
+      case 'addProperties' :
+        for (const key in char.extraData) {
+          obj[key] = char.extraData[key];
+        }
+        break;
+      case 'removeProperties' :
+        for (const key of char.keysToRemove) {
+          delete obj[key];
+        }
+        break;
+      case 'clear' :
+        for (const key in obj) {
+          delete obj[key];
+        }
+        break;
+    }
+
+    const clone = { ...obj };
+
+    arr.push(clone);
```

Comment:
```suggestion
    arr.push({ ...obj });
```

And [point 5](https://github.com/mate-academy/js_stateful-clones/blob/master/checklist.md)